### PR TITLE
langchain connectivity

### PR DESCRIPTION
**Pull request by moncapitaine**
closes #183

### CODE REVIEW

File: apps/chat-web/app/server-functions/langchain-chat-history.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { queryOptions } from '@tanstack/react-query'
+import { createServerFn } from '@tanstack/start'
+import { LangchainChatMessage, chatStore } from '../store/langchain-chat-store'
+
+const fetchChatHistory = createServerFn({ method: 'GET' }).handler(
+  async (): Promise<LangchainChatMessage[]> => {
+    // Simulate a slow network request
+    // await new Promise((resolve) => setTimeout(resolve, 2000))
```
Comment (YohannesTesfay) [inline apps/chat-web/app/server-functions/langchain-chat-history.ts:8]: Please remove.

File: apps/chat-web/app/store/langchain-chat-store.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+let langchainSessionId = (Math.random() + 1).toString(36).slice(7)
+
+const sendChatMessage = async (
+  message: string,
+): Promise<LangchainChatMessage[]> => {
+  const oldChat = await getChat()
+
+  // const answer =
+  //   possibleAnswers[Math.floor(Math.random() * possibleAnswers.length)]
```
Comment (YohannesTesfay) [inline apps/chat-web/app/store/langchain-chat-store.ts:39]: Clean up.

File: apps/chat-web/app/components/langchain-chat-form.tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  }
+
+  return (
+    <form onSubmit={handleSubmit} className="flex items-center gap-2">
+      <textarea
+        className="textarea textarea-bordered flex-grow"
+        name="message"
+        onKeyDown={handleTextareaKeyDown}
+      />
+      <button type="submit" className="btn btn-primary">
```
Comment (YohannesTesfay) [inline apps/chat-web/app/components/langchain-chat-form.tsx:84]: We could later disable the text area and send button until we receive a reply.
Future tickets.

**Review by YohannesTesfay — CHANGES_REQUESTED** [review]
Comments

File: apps/chat-web/app/server-functions/langchain-chat-history.ts (hunk 1 above)
Comment (moncapitaine) [reply apps/chat-web/app/server-functions/langchain-chat-history.ts]: done

**Pull request merged**